docs/endpoints/endpoints.rst

```diff
@@ -0,0 +1,743 @@
+
+``Strategy`` configuration is limited to two options:
+
+#. ``max_idletime``: Maximum duration in seconds that workers are allowed to idle before
+   they are marked for termination
+
+#. ``strategy_period``: Set the # of seconds between strategy attempting auto-scaling
+   events
+
+The bounds for scaling are determined by the options to the ``Provider``
+(``init_blocks``, ``min_blocks``, ``max_blocks``). Please refer to the `Parsl docs
+<https://parsl.readthedocs.io/en/stable/userguide/execution.html#elasticity>`_ for more
+info.
+
+Here's an example configuration:
+
+.. code-block:: yaml
+
+   engine:
+       type: GlobusComputeEngine
+       job_status_kwargs:
+           max_idletime: 60.0      # Default = 120s
+           strategy_period: 120.0  # Default = 5s
+
+
+Ensuring Execution Environment
+------------------------------
+
+When executing a function, endpoint *worker processes* expect to have all dependencies
+readily available.  For example, if a function requires ``numpy`` and a worker
+environment does not have that package installed, attempts to execute that function on
+that worker will fail.
+
+The process tree as shown in :ref:`starting the endpoint <endpoint-process-tree>` is the
+Compute Endpoint interchange.  This is distinct from the *worker* processes, which are
+managed by the Provider.  For example, the ProcessPoolEngine |nbsp| --- |nbsp| with a
+(conceptually) built‑in provider |nbsp| --- |nbsp| will create multiple new processes on
+the same host as the endpoint itself, whereas the GlobusComputeEngine might start
+processes (via the system‑specific batch scheduler) on entirely different hosts.  In
+:abbr:`HPC (High Performance Computing)` contexts, the latter is typically the case.
+
+As a result, it is often necessary to load in some kind of pre‑initialized environment
+for each worker.  In general there are two approaches:
+
+.. note::
+
+   The worker environment must have the ``globus-compute-endpoint`` Python module
+   installed.  We recommend matching the Python version and ``globus-compute-endpoint``
+   module version on the worker environment and on the endpoint interchange.
+
+1. Python-Based Environment Isolation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Python‑based environment management uses the |worker_init|_ config option:
+
+.. code-block:: yaml
+
+   engine:
+     provider:
+       worker_init: |
+         conda activate my-conda-env  # or venv, or virtualenv, or ...
+         source /some/other/config
+
+Though the exact behavior of ``worker_init`` depends on the specific |Provider|_, this
+is typically run in the same process as (or the parent process of) the worker, allowing
+environment modification (i.e., environment variables).
+
+In some cases, it may also be helpful to run some setup during the startup process of
+the endpoint itself, before any workers start.  This can be achieved using the top‑level
+``endpoint_setup`` config option:
+
+.. code-block:: yaml
+
+   endpoint_setup: |
```

Comment:
Could we make this config also show working_init being configured to activate the env for completeness?

Reply:
We do; look just above in the rendered documentation:

<img src="https://github.com/user-attachments/assets/4d712f48-66e7-4125-a43e-79e3eafd1006" title="" alt="" width="500">

Reply:
This looks obvious in the rendered form. No changes necessary.